t/db/005_hwsets.t

```diff
@@ -275,57 +255,62 @@ my $new_set7 = {
 	set_type   => 'HW',
 	set_params => { not_a_valid_field => 5 }
 };
-throws_ok {
-	$problem_set_rs->addProblemSet(
-		params => {
-			course_name => 'Precalculus',
-			%$new_set7
-		}
-	);
-}
-'DB::Exception::InvalidField', 'addProblemSet: adding an undefined parameter field';
+is(
+	dies { $problem_set_rs->addProblemSet(params => { course_name => 'Precalculus', %$new_set7 }); },
+	check_isa('DB::Exception::InvalidField'),
+	'addProblemSet: adding an undefined parameter field'
+);
 
 # Check for invalid parameter fields (the hide_hint param is a boolean)
-throws_ok {
-	$problem_set_rs->addProblemSet(
-		params => {
-			course_name => 'Precalculus',
-			set_name    => 'HW #11',
-			set_dates   => { open => 100, due => 140, answer => 200, enable_reduced_scoring => false },
-			set_type    => 'HW',
-			set_params  => { hide_hint => 'yes' }
-		}
-	);
-}
-'DB::Exception::InvalidParameter', 'addProblemSet: adding an non-valid parameter';
+is(
+	dies {
+		$problem_set_rs->addProblemSet(
+			params => {
+				course_name => 'Precalculus',
+				set_name    => 'HW #11',
+				set_dates   => { open => 100, due => 140, answer => 200, enable_reduced_scoring => false },
+				set_type    => 'HW',
+				set_params  => { hide_hint => 'yes' }
+			}
+		);
+	},
+	check_isa('DB::Exception::InvalidParameter'),
+	'addProblemSet: adding an non-valid parameter'
+);
 
 # Check to ensure true/false are passed into the set_params, not 0/1
-throws_ok {
-	$problem_set_rs->addProblemSet(
-		params => {
-			course_name => 'Precalculus',
-			set_name    => 'HW #11',
-			set_dates   => { open => 100, due => 140, answer => 200, enable_reduced_scoring => false },
-			set_type    => 'HW',
-			set_params  => { hide_hint => 0 }
-		}
-	);
-}
-'DB::Exception::InvalidParameter', 'addProblemSet: adding an non-valid boolean parameter';
+is(
+	dies {
+		$problem_set_rs->addProblemSet(
+			params => {
+				course_name => 'Precalculus',
+				set_name    => 'HW #11',
+				set_dates   => { open => 100, due => 140, answer => 200, enable_reduced_scoring => false },
+				set_type    => 'HW',
+				set_params  => { hide_hint => 0 }
+			}
+		);
+	},
+	check_isa('DB::Exception::InvalidParameter'),
+	'addProblemSet: adding an non-valid boolean parameter'
+);
 
 # Check to ensure true/false are passed into the enable_reduced_scoring in set_dates, not 0/1
-throws_ok {
-	$problem_set_rs->addProblemSet(
-		params => {
-			course_name => 'Precalculus',
-			set_name    => 'HW #11',
-			set_dates   => { open => 100, due => 140, answer => 200, enable_reduced_scoring => 0 },
-			set_type    => 'HW',
-			set_params  => { hide_hint => 0 }
-		}
-	);
-}
-'DB::Exception::InvalidParameter', 'addProblemSet: adding an non-valid boolean parameter in set_dates';
+is(
+	dies {
+		$problem_set_rs->addProblemSet(
+			params => {
+				course_name => 'Precalculus',
+				set_name    => 'HW #11',
+				set_dates   => { open => 100, due => 140, answer => 200, enable_reduced_scoring => 0 },
```

Comment:
`enable_reduced_scoring => false`

Reply:
Again, this needs to be 0.  That is the point of this test.

Reply:
Although, it seems that `set_params => { hide_hint => 0 }` should be `false` instead of `0` in this test so that the `enable_reduced_scoring` value is appropriately tested (the previous test already checks the `set_params` values.